[Maximus Tan] iP

maximustan2 opened the pull request:
## IpManPro: Your Task Management Solution

**Overview**

IpManPro is a simple yet powerful task management tool designed to free your mind from the burden of remembering things. It's easy to learn and incredibly fast to use. 😊

**Features**

* **Text-based:** IpManPro uses a straightforward text interface for easy input and management.
* **Fast and Efficient:** Experience lightning-fast performance and intuitive navigation.
* **Task Management:** Create, organize, and prioritize your tasks effortlessly.
* **Deadlines (Coming Soon):** Set due dates for tasks and stay on top of your commitments.
* **Reminders (Coming Soon):** Receive timely notifications to keep you focused and productive.

**Getting Started**

1. Download IpManPro from [here](https://github.com/miloaisdino/ip/releases).
2. Double-click the downloaded file to launch the application.
3. Add your tasks using the intuitive interface.
4. Let IpManPro manage your tasks for you, ensuring you never miss a deadline.

**Free and Open-Source**

IpManPro is completely free to use. If you're a Java programmer, you can even contribute to its development or use it as a learning tool.

**Example Usage**

```
java -jar ipman.jar
```
**Task List**

* Complete project proposal
* Schedule team meeting
* Research new marketing strategies

> "Your mind is for having ideas, not holding them." - David Allen

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

maximustan2 committed:
Change Duke to IpMan

maximustan2 committed:
feat: Echo commands and exit upon "bye"

maximustan2 committed:
feat: Add, List

maximustan2 committed:
feat: Mark as Done

maximustan2 committed:
refactor: Use switch statements instead

maximustan2 committed:
feat: ToDos, Events, Deadlines

maximustan2 committed:
feat: Automated Text UI Testing

maximustan2 committed:
feat: Error handling

maximustan2 committed:
refactor: Use ArrayList instead of array

maximustan2 committed:
feat: Ability to delete tasks

maximustan2 committed:
feat: Ability to recover from save state

maximustan2 committed:
fix: ignore data directory

maximustan2 committed:
Merge pull request #1 from miloaisdino/branch-Level-7

Level-7: Save

maximustan2 committed:
feat: Date parsing support

maximustan2 committed:
Merge pull request #2 from miloaisdino/branch-Level-8

feat: Date parsing support

maximustan2 committed:
refactor: OOPify the code

maximustan2 committed:
refactor: Organise into packages and update test cases

maximustan2 committed:
update: Readme.md

maximustan2 committed:
Merge pull request #3 from miloaisdino/add-gradle-support

Add gradle support

maximustan2 committed:
feat/refactor: Gradle pipeline added, package name renamed to lowercase, checkstyle added

maximustan2 committed:
test: Task.java

maximustan2 committed:
fix: data directory not found error

maximustan2 committed:
refactor: Add javadoc for methods and classes

maximustan2 committed:
refactor: Resolve checkstyle errors

maximustan2 committed:
Merge branch 'A-CodingStandard' into Level-9

To prevent merge conflicts later

maximustan2 committed:
feat: Give users a way to find a task by searching for a keyword.

maximustan2 committed:
Merge pull request #4 from miloaisdino/A-JavaDoc

Merge branch-A-JavaDoc into master

maximustan2 committed:
Merge pull request #5 from miloaisdino/A-CodingStandard

Merge A-CodingStandard into master

maximustan2 committed:
Merge pull request #6 from miloaisdino/Level-9

Merge Level-9 into master

tingxuanp commented on src/main/java/ipman/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+    protected LocalDateTime by;
+
+    /**
+     * Constructor
+     * @param description Name of deadline
+     * @param stringDate String in DD/MM/YYYY hh[0-23]mm format
+     * @throws CommandException Error
```
This comment applies for all your Javadoc comments that violate this coding standard. There should be an empty line between the description and parameter section.

tingxuanp commented on src/main/java/ipman/Parser.java:
```diff
@@ -0,0 +1,160 @@
+ * @author miloaisdino
+ */
+public class Parser {
+    private final ArrayList<Task> list;
```
Since this is an ArrayList of Tasks, perhaps you could have named it in plural form, such as `tasks` instead to clearly distinguish that this variable is a collection of multiple items?

tingxuanp commented on src/main/java/ipman/IpMan.java:
```diff
@@ -0,0 +1,54 @@
+        }
+    }
+
+    /**
+     *
+     */
```
Is there supposed to be a method description here?

tingxuanp commented on src/main/java/ipman/Task.java:
```diff
@@ -0,0 +1,45 @@
+     * Sets the task to not done
+     * @return The new string representation of the task
+     */
+    public String unmarkStatus() {
```
It is great that your method names are able to explain the methods.

maximustan2 committed:
feat: Refactor JavaDocs

maximustan2 committed:
feat: Refactor variable name of list

maximustan2 committed:
feat: Implement checkstyle

maximustan2 committed:
Merge pull request #7 from miloaisdino/branch-A-CheckStyle

feat: Implement checkstyle

maximustan2 committed:
fix: Refactor JavaDoc of flushBuffer

JellyPenguinnn commented on src/main/java/ipman/Todo.java:
```diff
@@ -0,0 +1,24 @@
+ *
+ * @author miloaisdino
+ */
+public class Todo extends Task {
```
Good use of inheritance, no major issues related to coding standards

JellyPenguinnn commented on src/main/java/ipman/Parser.java:
```diff
@@ -0,0 +1,164 @@
+                }
+            } catch (CommandException ce) {
+                ui.addToBuffer(ce.getMessage());
+            } catch (Exception e) {
```
Catch more specific exceptions where possible to improve error diagnosis

JellyPenguinnn commented on src/main/java/ipman/Storage.java:
```diff
@@ -0,0 +1,81 @@
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            //do nothing
```
Empty catch block should be avoided

JellyPenguinnn reviewed (commented):
The codes are well-structured with clear JavaDoc comments. 
Solid understanding of Java programming principles and object-oriented design.
Great work, and keep it up!!

maximustan2 committed:
feat: Implement GUI for IpMan

maximustan2 committed:
Merge branch 'master' into branch-Level-10

maximustan2 committed:
Merge pull request #8 from miloaisdino/branch-Level-10

feat: Implement GUI for IpMan

maximustan2 committed:
fix: Handle file permissions errors

maximustan2 committed:
refactor: Header comments for main()

Improve JavaDocs for main method

This method forms the entrypoint of the application.

Refactoring the JavaDocs for this method improves code quality.

Let’s rewrite the JavaDoc for this method.

maximustan2 committed:
refactor: Move command logic to a separate class to preserve SLAP

maximustan2 committed:
refactor: Move command logic to a separate class to preserve SLAP

The parseLine method in Parser is too long.

Let's break the method down into multiple methods to reduce each method length.

maximustan2 committed:
Merge remote-tracking branch 'origin/A-CodeQuality' into A-CodeQuality

maximustan2 committed:
Merge pull request #10 from miloaisdino/A-CodeQuality

A-CodeQuality

maximustan2 committed:
feat: Ability to tag individual tasks

maximustan2 committed:
Merge pull request #11 from miloaisdino/branch-C-Tagging

feat: Ability to tag individual tasks

maximustan2 committed:
Merge branch 'master' into branch-A-Assertions

maximustan2 committed:
test: Add runtime assertions to Storage class

maximustan2 committed:
Merge pull request #12 from miloaisdino/branch-A-Assertions

A-Assertions

maximustan2 committed:
feat: Change the images of users

maximustan2 committed:
docs: Complete coverage for JavaDocs

maximustan2 committed:
docs: Upload screenshot of app

maximustan2 committed:
docs: Add user guide

maximustan2 committed:
docs: Add name to CONTRIBUTORS.md

maximustan2 committed:
test: Resolve E2E test errors

maximustan2 committed:
test: Implement CI on Github actions
